File: Linguagem C Exercicios/Matrizes/SOMA LINHAS.c
#include <stdio.h>
#include <stdlib.h>

int main()
{

    int m,n;

    printf("Qual a quantidade de linhas da matriz? ");
    scanf("%d",&m);
    printf("Qual a quiantidade de colunas da matriz? ");
    scanf("%d",&n);

    double mat[m][n];
    double vet[m];

    for (int i=0;i<m;i++){
           printf("Digite os elementos da %da. linha:\n",i+1);
        for(int j=0;j<n;j++){
            scanf("%lf",&mat[i][j]);
        };
    };

    for (int i=0;i<m;i++){
        vet[i] =0;
        for (int j = 0;j<n;j++){
            vet[i] = vet[i]+mat[i][j];
        };
    };

    printf("VETOR GERADO:\n");
    for (int i=0;i<m;i++){
        printf("%.1lf\n",vet[i]);
    };


    return 0;
}

File: Linguagem C#/Matrizes/SomaLinhas.cs
using System;
using System.Globalization;

namespace SomaLinhas {
    class Program {
        static void Main(string[] args) {

            CultureInfo CI = CultureInfo.InvariantCulture;

            int m, n;
            double soma;

            Console.Write("Qual a quantidade de linhas da matriz? ");
            m = int.Parse(Console.ReadLine());
            Console.Write("Qual a quantidade de colunas da matriz? ");
            n = int.Parse(Console.ReadLine());

            double[,] mat = new double[m, n];
            double[] vet = new double[m];

            for (int i = 0; i < m; i++) {
                Console.WriteLine("Digite os elementos da "+(i+1)+"a linha:");
                for (int j = 0; j < n; j++) {
                    mat[i, j] = double.Parse(Console.ReadLine(),CI);
                };
            };

            Console.WriteLine("VETOR GERADO:");
            for (int i = 0; i < m; i++) {
                soma = 0;
                for (int j = 0; j < n; j++) {
                    soma = soma + mat[i, j];
                    vet[i] = soma;                    
                };               
            };

            for (int i = 0; i < m; i++) {
                Console.WriteLine(vet[i].ToString("F1",CI));
            };
            
        }
    }
}
